### DISCUSSION TITLE
Jobs clarification

### DISCUSSION THREAD

**Discussion by wesoos** [shinyorg/shiny#1379] [Q&A] [answered]
Hello, when registering a job, and setting the RunOnForeground to true, does it only run in the foreground now, or still in the background as well?  The reason I'm asking is my app is in the background and I have not seen the job run.  It's been more than 30 minutes..

Thanks

**Comment by aritchie** [comment, marked as answer]
Jobs will run, they are periodic NOT scheduled.  iOS can be 1 day+ before it starts firing.  It tries to respond based on user usage.  Android jobs are far more regular within a few hours.  

RunOnForeground just means that it will be triggered every 30 seconds while your app is live.  

If you think the job isn't running, you can use a local logger to review what may be happening
